pkg/observability/stackdriver.go

```diff
@@ -121,12 +125,21 @@ func (e *stackdriverExporter) StartExporter(ctx context.Context) error {
 			Name:             fmt.Sprintf("projects/%s", e.config.ProjectID),
 			MetricDescriptor: md,
 		}
-		_, err = mclient.CreateMetricDescriptor(ctx, cmrdesc)
-		if err != nil {
-			return fmt.Errorf("failed to create MetricDescriptor: %w", err)
-		}
+		descriptorCreateRequests = append(descriptorCreateRequests, cmrdesc)
 	}
 
+	// register metrics in the background and don't block server startup on failures.
+	go func() {
+		logger.Infow("starting metric registration")
+		for _, cmrdesc := range descriptorCreateRequests {
+			_, err = mclient.CreateMetricDescriptor(ctx, cmrdesc)
```

Comment:
I think this `ctx` has a timeout from the parent. I think we should create a new per-request context:

```go
ctx, done := context.WithTimeout(context.Background(), 10*time.Second)
defer done()
```

If we want to limit the overall time the function can run:

```go
go func() {
  ctx, done := context.WithTimeout(context.Background(), 30*time.Second)
  defer done()


  for _, cmrdesc := range descriptorCreateRequests {
    func() {
      subCtx, done := context.WithTimeout(ctx, 5*time.Second)
      defer done()

      if _, err := mclient.CreateMetricDescriptor(subCtx, cmrdesc); err != nil {
        logger.Errorw(...)
      }
    }()
  }
}()
```

Reply:
good catch - updated